### PR TITLE
Week 19 - Darcy

### PR DESCRIPTION
**Pull request by darcywinterhill**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+POST '/thougths'
+
+Lets you post a new message to the backend.
+
+---
+
+POST '/thoughts/:thoughtId/like'
+
+Increases nr of likes/hearts.
+
```
Comment (Aepera) [inline README.md:29]: Good descriptions of endpoints.

File: server.js [hunk 2]
```diff
@@ -1,29 +1,98 @@
...
+    res.status(200).json(like)
+    } else {
+      res.status(404).json({
+        message: 'Message not found'
+      })
+    }
+  } catch (error) {
+    res.status(400).json({
+      message: 'Invalid request',
+      error
```
Comment (Aepera) [inline server.js:91]: Nice and clear error handling 👍

File: server.js [hunk 3]
```diff
@@ -1,29 +1,98 @@
...
+const port = process.env.PORT || 9000
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+ message: {
+   type: String,
+   required: [true, 'Message is required'],
+   minlength: [5, 'Minimum length is 5 characters'],
+   maxlength: [140, 'Maximum length is 140 characters']
```
Comment (Aepera) [inline server.js:21]: Great sending some extra information about the requirements.

File: server.js [hunk 4]
```diff
@@ -1,29 +1,98 @@
...
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({
+      message: 'Service unavailable'
+    })
```
Comment (Aepera) [inline server.js:44]: I guess this is  middleware handling if server is unavailable and sending error message? Very nice adding this to the app. 🎉

**Review by Aepera — APPROVED** [review]
Great work! Cannot find anything to not approve 👌👸✔

File: server.js (hunk 3 above)
Comment (KarolCislowski) [inline server.js:21]: great configured field 🥳

File: server.js [hunk 5]
```diff
@@ -1,29 +1,98 @@
...
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+//endpoint to GET a list of happy thoughts
+app.get('/thoughts', async (req, res) => {
+    const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
```
Comment (KarolCislowski) [inline server.js:54]: 💯

File: server.js [hunk 6]
```diff
@@ -1,29 +1,98 @@
...
+
+//endpoint to increase nr of likes/hearts
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params
+
+  try {
+    const like = await Thought.findOneAndUpdate(
+      { _id: thoughtId },
+      { $inc: { hearts: 1 } },
+      { new: true }
```
Comment (KarolCislowski) [inline server.js:79]: Nice 👍🏻

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Your code looks nice but You need to provide a link to the hosted frontend connected to this API. 😸

**Review by KarolCislowski — APPROVED** [review]
Great Project!

**Pull request closed without merging**